File: code/js/src/domain/game/GameModel.ts
import { User } from '../user/User';
import { Color } from './Color';
import { GameState } from './GameState';

export class GameModel {
  constructor(
    public state: GameState,
    public black: User,
    public white: User,
    public link: string,
    public opponent: Color | null
  ) {}
}

File: code/jvm/src/main/kotlin/pt/isel/daw/gomoku/services/models/GameModel.kt
package pt.isel.daw.gomoku.services.models

import pt.isel.daw.gomoku.domain.game.Color
import pt.isel.daw.gomoku.domain.game.GameState
import pt.isel.daw.gomoku.domain.user.User

data class GameModel(
    val gameId: Int,
    val state: GameState,
    val black: User,
    val white: User,
    val opponent: Color?
)
